aws/s3/s3_integration_test.go

```diff
@@ -439,6 +481,56 @@ func TestS3List(t *testing.T) {
 	}
 }
 
+func TestS3ListAll(t *testing.T) {
+	// ARRANGE
+	setup()
+	defer teardown()
+
+	// populate bucket with over 1000 objects (the limit at
+	// which ListAll's continuation token functionality is required).
+	numObjects := 1005
+	keys := make([]string, 0)
+	keyGroups := [11]string{"A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K"}
+	for i := 0; i < numObjects; i++ {
+		keyGroup := keyGroups[i/100]
+		keys = append(keys, fmt.Sprintf("%s%s%s%06d", testPrefix, keyGroup, testPrefixDelimiter, i))
+	}
+	awsCmdPutKeys(keys)
+
+	client, err := New()
+	require.Nil(t, err, fmt.Sprintf("Error creating s3 client: %v", err))
+
+	// ACTION
+	listing, err := client.ListAll(testBucket, testPrefix)
+
+	// ASSERT
+
+	// got listing ok
+	assert.Nil(t, err)
+
+	// expected objects in list in the correct order.
+	assert.Equal(t, listing, keys)
+
+	// ACTION
+	testPrefixes := make([]string, 0)
+	for _, keyGroup := range keyGroups {
+		testPrefixes = append(testPrefixes, testPrefix+keyGroup)
+	}
+	listingConcurrent, err := client.ListAllObjectsConcurrently(testBucket, testPrefixes)
+	var keysConcurrent []string
+	for _, object := range listingConcurrent {
+		keysConcurrent = append(keysConcurrent, aws.ToString(object.Key))
```

Comment:
You don't want the import in the test file? I can remove if you want since the test should still pass, just did it from habit to avoid chance of null.

Reply:
Hmmm on second thoughts, keep it as is...next time I'm in the codebase, I'll probably change instances of `*object.Key` should change `aws.ToString(object.Key)` like you have.